google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/CloudSdkAppEngineHelper.java

```diff
@@ -143,18 +144,26 @@ public Project getProject() {
           deploy, isFlexCompat));
     } else if (targetEnvironment.isFlexible()) {
       try {
-        // Checks if the Yaml or Dockerfile exist.
-        Optional<FlexibleRuntime> runtimeOptional =
-            AppEngineProjectService.getInstance().getFlexibleRuntimeFromAppYaml(
-                deploymentConfiguration.getAppYamlPath());
+        AppEngineFlexibleFacet flexFacet =
+            AppEngineFlexibleFacet.getFacetByModuleName(
+                deploymentConfiguration.getModuleName(), project);
 
-        if (!Files.exists(Paths.get(deploymentConfiguration.getAppYamlPath()))) {
+        if (flexFacet == null
+            || !Files.exists(Paths.get(flexFacet.getConfiguration().getAppYamlPath()))) {
```

Comment:
for the second case, we get the following error printed in the deploy console:

`Failed to deploy '[2017-06-09 13:31:47] demo:war. Project: test-default-1344. Version: asdf': The specified app.yaml configuration file does not exist or is not a valid file.`

I think that is reasonable. I don't know of a way (or even if its desireable) to dynamically invalidate a run config after updating something in the facet.

For 1 - you are totally right..I feel silly for not thinking of that. I wish there was something more resilient that we could key off of. If not, then your suggestion is probably a good one.